src/GitVersionCore/Extensions/GitExtensions.cs

```diff
@@ -43,24 +43,24 @@ public static bool IsSameBranch(this Branch branch, Branch otherBranch)
         /// <summary>
         /// Exclude the given branches (by value equality according to friendly name).
         /// </summary>
-        public static IEnumerable<BranchCommit> ExcludingBranches(this IEnumerable<BranchCommit> branches, IEnumerable<Branch> branchesToExclude)
+        public static IEnumerable<BranchCommit> ExcludingBranches(this IEnumerable<BranchCommit> branches, IEnumerable<IBranch> branchesToExclude)
         {
             return branches.Where(b => branchesToExclude.All(bte => !IsSameBranch(b.Branch, bte)));
         }
 
         /// <summary>
         /// Exclude the given branches (by value equality according to friendly name).
         /// </summary>
-        public static IEnumerable<Branch> ExcludingBranches(this IEnumerable<Branch> branches, IEnumerable<Branch> branchesToExclude)
+        public static IEnumerable<IBranch> ExcludingBranches(this IEnumerable<IBranch> branches, IEnumerable<IBranch> branchesToExclude)
         {
             return branches.Where(b => branchesToExclude.All(bte => !IsSameBranch(b, bte)));
         }
-        public static IEnumerable<Commit> CommitsPriorToThan(this Branch branch, DateTimeOffset olderThan)
+        public static IEnumerable<ICommit> CommitsPriorToThan(this IBranch branch, DateTimeOffset olderThan)
```

Comment:
Any reason not to provide `CommitsPriorToThan` as a method on the `IBranchCollection` interface?

Reply:
Agree, I will change that in a subsequent PR